test/inductor/test_torchinductor.py

```diff
@@ -4505,6 +4505,23 @@ def fn(a, b):
 
         self.common(fn, [torch.randn(10, 1024), torch.randn(10, 512)])
 
+    def test_as_strided_scatter_on_input_tensor_with_storage_offset(self):
+        def fn(input, src):
+            return aten.as_strided_scatter(
+                input,
+                src,
+                size=input.size(),
+                stride=input.stride(),
+                storage_offset=1,
+            )
+
+        inp = torch.randn(4, 1)
+        # as_strided_scatter in inductor is only making a clone of input
+        # and doesn't copy from src, clone src to itself for testing
```

Comment:
It might be worth opening a separate issue. The copy is definitely in the lowering, so it would suggest inductor isn't treating the mutation properly.

Reply:
@yanboliang mentioned it might be intended to save operation as _as_strided_scatter_meta only need the output's tensor size info

Reply:
Actually what we are trying to fix is a prims bug rather than inductor bug, you can put the test in prims unit test folder. But since you found the inductor unit test bug, so I'm ok if you put it here. But your comment is not correct, you described what prims is doing, not what inductor is doing.